river/Seat.zig

```diff
@@ -60,6 +60,12 @@ mode_id: usize = 0,
 /// ID of previous keymap mode, used when returning from "locked" mode
 prev_mode_id: usize = 0,
 
+/// Timer for repeating keyboard mappings
+mapping_repeat_timer: *wl.EventSource = undefined,
+
+/// Currently repeating mapping, if any
+mapping_repeat_command_args: ?[]const [:0]const u8 = null,
```

Comment:
I originally tried it that way but was getting crashes dereferencing the saved Mapping. From what I see, the Mapping is owned by an ArrayList which does not guarantee that it will remain at a stable address. I was going to copy the Mapping and save that, but the Mapping was already creating an owned copy of the command args. Borrowing that seemed simpler.

Reply:
Ah, of course.

You could change it to use something other than an ArrayList, like a TailQueue or SinglyLinkedList. I think it would be worth it because it makes the repeat code a bit cleaner IMO.

Reply:
Turns out it was my inexperience with Zig... in `for (...mappings.items) |mapping| { }` `mapping` is a temporary copy of the Mapping, not a reference to it. I was storing a reference to the temp, which obviously won't work. I've added a fix using `for (...) |*mapping| { }` (which seems like it would be more efficient anyway).